src/main/java/core/basesyntax/strategy/impl/SupplyOperation.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        if (transaction == null) {
+            throw new RuntimeException("Transaction cannot be null.");
+        }
+        int currentQuantity = Storage.getFruitQuantity(transaction.getFruit());
+        int newQuantity = currentQuantity + transaction.getQuantity();
```

You use `transaction.getQuantity()` at line 14 without validating it. Even though DataConverter enforces non-negative quantities, handlers should be defensive. Add a check such as `if (transaction.getQuantity() < 0) { throw new RuntimeException("Quantity cannot be negative: " + transaction.getQuantity()); }` before computing newQuantity so invalid transactions fail fast with an informative message.